src/base/bittorrent/session.cpp

```diff
@@ -1285,36 +1302,65 @@ void Session::populateAdditionalTrackers()
     }
 }
 
-void Session::processBigRatios()
+void Session::processShareLimits()
 {
-    qDebug("Process big ratios...");
+    qDebug("Processing share limits...");
 
-    qreal globalMaxRatio = this->globalMaxRatio();
     foreach (TorrentHandle *const torrent, m_torrents) {
         if (torrent->isSeed()
-            && (torrent->ratioLimit() != TorrentHandle::NO_RATIO_LIMIT)
             && !torrent->isForced()) {
-            const qreal ratio = torrent->realRatio();
-            qreal ratioLimit = torrent->ratioLimit();
-            if (ratioLimit == TorrentHandle::USE_GLOBAL_RATIO) {
-                // If Global Max Ratio is really set...
-                ratioLimit = globalMaxRatio;
-                if (ratioLimit < 0) continue;
-            }
-            qDebug("Ratio: %f (limit: %f)", ratio, ratioLimit);
-            Q_ASSERT(ratioLimit >= 0.f);
-
-            if ((ratio <= TorrentHandle::MAX_RATIO) && (ratio >= ratioLimit)) {
-                Logger* const logger = Logger::instance();
-                if (maxRatioAction() == Remove) {
-                    logger->addMessage(tr("'%1' reached the maximum ratio you set. Removing...").arg(torrent->name()));
-                    deleteTorrent(torrent->hash());
+            if (torrent->ratioLimit() != TorrentHandle::NO_RATIO_LIMIT) {
+                const qreal ratio = torrent->realRatio();
+                qreal ratioLimit = torrent->ratioLimit();
+                if (ratioLimit == TorrentHandle::USE_GLOBAL_RATIO)
+                    // If Global Max Ratio is really set...
+                    ratioLimit = this->globalMaxRatio();
```

Comment:
True, that's the result of the search/replace from the previous implementation that used a global variable instead.  I'll fix it.